lib/datasets_test.go

```diff
@@ -384,103 +385,118 @@ func TestDatasetRequestsGet(t *testing.T) {
-		{&GetParams{Selector: "body", Format: "json", Path: clRef.String(), Limit: 2, Offset: 0, All: false}, 2, ""},
-		{&GetParams{Selector: "body", Format: "json", Path: sitemapRef.String(), Limit: 3, Offset: 0, All: false}, 3, ""},
+		{&GetParams{Path: "peer/movies", Selector: "commit"},
+			componentToString(moviesDs.Commit, "yaml")},
+
+		{&GetParams{Path: "peer/movies", Selector: "commit", Format: "json"},
+			componentToString(moviesDs.Commit, "json")},
+
+		{&GetParams{Path: "peer/movies", Selector: "commit.title"}, "initial commit\n"},
+
+		{&GetParams{Path: "peer/movies", Selector: "commit.title", Format: "json"}, "\"initial commit\""},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "json"}, "[]"},
+
+		{&GetParams{Path: "", Selector: "body", Format: "json"}, "repo: empty dataset reference"},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "csv"}, "title,duration\n"},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "json",
+			Limit: 5, Offset: 0, All: false}, bodyToString(moviesBody[:5])},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "json",
+			Limit: -5, Offset: -100, All: false}, "invalid limit / offset settings"},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "json",
+			Limit: -5, Offset: -100, All: true}, bodyToString(moviesBody)},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "json",
+			Limit: 0, Offset: 0, All: true}, bodyToString(moviesBody)},
+
+		{&GetParams{Path: "peer/movies", Selector: "body", Format: "json",
+			Limit: 2, Offset: 10, All: false}, bodyToString(moviesBody[10:12])},
 	}
 
 	req := NewDatasetRequests(node, nil)
 	for i, c := range cases {
 		got := &GetResult{}
-		err := req.Get(c.p, got)
-
-		if !(err == nil && c.err == "" || err != nil && err.Error() == c.err) {
-			t.Errorf("case %d error mismatch: expected: %s, got: %s", i, c.err, err)
+		err := req.Get(c.params, got)
+		if err != nil {
+			if err.Error() != c.expect {
+				t.Errorf("case %d error mismatch: expected: %s, got: %s", i, c.expect, err)
+			}
 			continue
 		}
 
-		if got.Bytes == nil && c.resCount == 0 {
-			continue
+		result := string(got.Bytes)
+		if result != c.expect {
+			t.Errorf("case %d Bytes mismatch: expected:\n\"%s\", got:\n\"%s\"", i, c.expect, result)
 		}
+	}
+}
 
-		switch c.p.Format {
-		default:
-			// default should be json format
-			_, err := json.Marshal(got.Dataset)
-			if err != nil {
-				t.Errorf("case %d error parsing response data: %s", i, err.Error())
-				continue
-			}
-		case "csv":
-			r := csv.NewReader(bytes.NewBuffer(got.Bytes))
-			_, err := r.ReadAll()
-			if err != nil {
-				t.Errorf("case %d error parsing response data: %s", i, err.Error())
-				continue
-			}
+func setDatasetName(ds *dataset.Dataset, name string) *dataset.Dataset {
+	parts := strings.Split(name, "/")
+	ds.Peername = parts[0]
+	ds.Name = parts[1]
+	return ds
+}
+
+func componentToString(component interface{}, format string) string {
```

Comment:
just thinking out loud here, but `ComponentToString` might make a nice exported function on `github.com/qri-io/dataset/dstest`. If we come across a second use for this let's do it.